sympy/holonomic/holonomic.py

```diff
@@ -2008,10 +2008,10 @@ def to_hyper(self, as_list=False, _recur=None):
         b = r.listofpoly[-1]
 
         # the constant multiple of argument of hypergeometric function
-        if isinstance(a.rep[0], (PolyElement, FracElement)):
-            c = - (S(a.rep[0].as_expr()) * m**(a.degree())) / (S(b.rep[0].as_expr()) * m**(b.degree()))
+        if isinstance(a.to_list()[0], (PolyElement, FracElement)):
+            c = - (S(a.to_list()[0].as_expr()) * m**(a.degree())) / (S(b.to_list()[0].as_expr()) * m**(b.degree()))
         else:
-            c = - (S(a.rep[0]) * m**(a.degree())) / (S(b.rep[0]) * m**(b.degree()))
+            c = - (S(a.to_list()[0]) * m**(a.degree())) / (S(b.to_list()[0]) * m**(b.degree()))
```

Comment:
Something like this can be more efficient:
```python
In [33]: from sympy.polys.polyclasses import DMP

In [34]: p = DMP([1, 2, 3], ZZ)

In [35]: p
Out[35]: DUP_Flint([1, 2, 3], ZZ)

In [36]: p.to_list()[0]
Out[36]: 1

In [37]: p.nth(0)
Out[37]: 3

In [38]: p.nth(2)
Out[38]: 1

In [39]: p.nth(p.degree() - 0)
Out[39]: 1
```

Reply:
I doubt that it matters much though and it is not always trivial to know what the code wants to do in the multivariate case.

Reply:
Actually it should just be LC:
```python
In [41]: p.to_list()[0]
Out[41]: 1

In [42]: p.LC()
Out[42]: 1
```

Reply:
I've changed this to use `.LC`